### PR TITLE
Nevin/bug/4942

### PR DESCRIPTION
**Pull request by cnevinc**

### CODE REVIEW

**Comment by cnevinc** [comment]
At the second thought, maybe we just need Dagger in Java instead of make it all Kotlin?

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.kt [hunk 1]
```diff
@@ -300,7 +300,7 @@ class MainActivity : BaseActivity(),
             showTabTray.observe(this@MainActivity, Observer {
                 val tabTray = TabTray.show(supportFragmentManager)
                 if (tabTray != null) {
-                    tabTray.setOnDismissListener { portraitStateModel.cancelRequest(PortraitComponent.TabTray) }
+                    tabTray.setOnDismissListener(DialogInterface.OnDismissListener { portraitStateModel.cancelRequest(PortraitComponent.TabTray) })
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:303]: is this necessary?

File: app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt [hunk 2]
```diff
@@ -2,847 +2,721 @@
...
-    private static final boolean ENABLE_BACKGROUND_ALPHA_TRANSITION = true;
-    private static final boolean ENABLE_SWIPE_TO_DISMISS = true;
+        @JvmStatic
+        fun newInstance(): TabTrayFragment? {
+            return TabTrayFragment()
+        }
+    }
 
-    private static final float OVERLAY_ALPHA_FULL_EXPANDED = 0.50f;
+    private var presenter: TabTrayContract.Presenter? = null
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt:89]: we could make it lateinit var

File: app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt [hunk 3]
```diff
@@ -2,847 +2,721 @@
...
-    private Drawable backgroundOverlay;
+    private var recyclerView: ThemedRecyclerView? = null
+    private var layoutManager: LinearLayoutManager? = null
 
-    private ThemedRecyclerView recyclerView;
-    private LinearLayoutManager layoutManager;
+    private var playEnterAnimation = true
 
-    private boolean playEnterAnimation = true;
+    private var adapter: TabTrayAdapter? = null
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt:109]: we could make it lateinit var, then we don't need the null check

File: app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt [hunk 4]
```diff
@@ -2,847 +2,721 @@
...
 
-    private Handler uiHandler = new Handler(Looper.getMainLooper());
+    private val slideCoordinator: SlideAnimationCoordinator = SlideAnimationCoordinator(this)
 
-    private SlideAnimationCoordinator slideCoordinator = new SlideAnimationCoordinator(this);
+    private val dismissRunnable = Runnable { dismissAllowingStateLoss() }
 
-    private Runnable dismissRunnable = this::dismissAllowingStateLoss;
+    private var tabTrayViewModel: TabTrayViewModel? = null
+    private var homeViewModel: HomeViewModel? = null
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt:118]: we could make it lateinit var, then we don't need the null check

File: app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt [hunk 5]
```diff
@@ -2,847 +2,721 @@
...
+    private val uiHandler = Handler(Looper.getMainLooper())
 
-    private Handler uiHandler = new Handler(Looper.getMainLooper());
+    private val slideCoordinator: SlideAnimationCoordinator = SlideAnimationCoordinator(this)
 
-    private SlideAnimationCoordinator slideCoordinator = new SlideAnimationCoordinator(this);
+    private val dismissRunnable = Runnable { dismissAllowingStateLoss() }
 
-    private Runnable dismissRunnable = this::dismissAllowingStateLoss;
+    private var tabTrayViewModel: TabTrayViewModel? = null
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt:117]: we could make it lateinit var, then we don't need the null check

File: app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt [hunk 6]
```diff
@@ -2,847 +2,721 @@
...
 
-    private ThemedImageView imgPrivateBrowsing, imgNewTab;
-    private ThemedView bottomDivider;
+    private var imgPrivateBrowsing: ThemedImageView? = null
+    private var imgNewTab: ThemedImageView? = null
+    private var bottomDivider: ThemedView? = null
 
-    private ShoppingSearchItemDecoration itemDecoration;
-    private boolean showShoppingSearch;
+    private var itemDecoration: ShoppingSearchItemDecoration? = null
```
Comment (proevan) [inline app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.kt:127]: itemDecoration

**Comment by benjamin-cheng** [comment]
Close this issue since we decided to kotlinize the TabTrayFragment in the separate PR. And I already has the PR #4946 to fix the dagger issue.

**Pull request closed without merging**